pdns/lua-record.cc

```diff
@@ -255,6 +277,9 @@ bool IsUpOracle::isUp(const CheckDesc& cd)
     auto iter = statuses->find(cd);
     if (iter != statuses->end()) {
       iter->second->lastAccess = now;
+      if (iter->second->weight > 0) {
```

Comment:
This logic means that, if an ip is found up with a non-zero weight, and then later gets marked down (`status` = `false`), it would still get returned as up with its previous weight.

You need to either check for `status` first, or reset `weight` to `0` in `setDown`.

Reply:
added weight reset when a candidate is down